packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -493,10 +494,11 @@ export function model<T = {}>(properties?: IModelProperties<T>): IModelType<Snap
  * @export
  * @alias types.model
  */
-export function model(...args: any[]) {
-    const name = typeof args[0] === "string" ? args.shift() : "AnonymousModel"
+export function model<T = {}>(...args: any[]): IModelType<T | Snapshot<T>, T> {
+    const name = typeof args[0] === "string" ? (args.shift() as string) : "AnonymousModel"
     const properties = args.shift() || {}
-    return new ModelType({ name, properties })
```

Comment:
This and above fixes the error: `src/types/complex-types/model.ts(483,17): error TS2394: Overload signature is not compatible with function implementation.`